### DISCUSSION TITLE
ArrayField动态值问题

### DISCUSSION THREAD

**Discussion by VinVC** [alibaba/formily#2989] [Q&A 2.x]
### 场景：
```js
{
  taskType: 1,
  strategyList: [
    {
      ratio: 0, // 用户占比
      code: '', // 策略编号
      name: '', // 策略名称
      actionList: [
        {
          requiredNum: 1, // 要求的次数
          rewardType: 1, // 奖励形式（1现金，2积分）
          cashAmount: 120.12, // 奖励现金金额
          scoreNum: 40, //奖励积分值
        {
      ]
    }
  ]
}
```
taskType === 1： actionList 可自增；
taskType === 2： actionList 不可自增，长度为2, 且actionList[0]和actionList[1]的rewardType的addonBefore文案不同
### 问题：
如何在taskType变化的时候设置actionList的值？actionList Field.setInitialValue 和 actionList field.setValue已尝试均不行。setInitialValue导致strategyList无法删除子项；setValue会导致报错；
[codesandbox demo](https://codesandbox.io/s/clever-dewdney-6e0cw4?file=/App.tsx)

**Comment by janryWang** [comment]
具体问题的复现步骤是什么

**Reply by VinVC** [reply]
根据你在[这个问题](https://github.com/alibaba/formily/discussions/2987)里的回答，改成了主动模式，然后再调用setValue就不报错了。谢谢。
请问一下，为什么ArrayField 的 setValue在主被动中会有这种差异？比如：a是个ArrayField
- 被动：onFieldReact("a", fun)，  fun中依赖了b，当b变化时，`a.setValue`会报错
- 主动：onFieldInputValueChange("b", fun)，当b变化时，  fun中`a.setValue`就不会报错
